segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/delegationtoken/TokenVerifierImpl.java

```diff
@@ -75,6 +75,14 @@ public JsonWebToken verifyToken(@NonNull String resource, String token, @NonNull
      * @return
      */
     private boolean resourceMatchesClaimKey(String claimKey, String resource) {
+        log.trace("claimKey = {}, resourceKey = {}", claimKey, resource);
+        if (claimKey.startsWith("prn::")) {
+            claimKey = claimKey.replace("prn::/", "")
```

Comment:
Since you know your replaced string is at the beginning, use `substring`. Much faster to process.

Reply:
IMO, the code looks simpler with the `replace` call and reveals its intention better. 

Besides, resources can be both of the form `prn::/...` and `prn::*`, which is why I'm replacing the former first and then attempting to replace `prn::`.

Reply:
I have modified the code with a much more efficient logic. Please check out the new logic.